ycmd.el

```diff
@@ -610,26 +610,30 @@ control.) The newly started server will have a new HMAC secret."
 
   (ycmd--start-keepalive-timer))
 
+(defun ycmd--shutdown ()
+  "Send a shutdown request to the server."
+  (deferred:sync!
+    (ycmd--request "/shutdown" nil :parser 'json-read)))
+
 (defun ycmd-close (&optional time-out-secs)
   "Shutdown any running ycmd server.
 
-Wait TIME-OUT-SECS seconds after `interrupt-process' call for the
-ycmd server to end before killing the process with
-`delete-process'.
+Wait TIME-OUT-SECS seconds after a shutdown request for the ycmd
+server to end before killing the process with `delete-process'.
 
 This does nothing if no server is running."
   (interactive)
 
-  (unwind-protect
```

Comment:
Good question. I guess that either a) it's a vestige that can be removed or b) ycmd--global-teardown was supposed to be the unwind form. Maybe it should be `(when (ycmd-running?) (unwind-protect (...shutdown...) (ycmd--global-teardown)))`. But I'm not sure how it got this way...